datalad/interface/annotate_paths.py

```diff
@@ -243,7 +245,7 @@ def get_modified_subpaths(aps, refds, revision, recursion_limit=None,
                 ap.update(m)
                 yield ap
                 break
-            if m['path'].startswith(_with_sep(ap['path'])):
+            if path_startswith(m['path'], ap['path']):
```

Comment:
Not sure whether this actually is a bugfix, but here the outcome is different from before, if both paths are equal.